### PR TITLE
fix incorrect setupBindings prop names and $realm

### PR DESCRIPTION
**Pull request by Kingwl**
fix issue #1235

### CODE REVIEW

**Comment by Kingwl** [comment]
i have no idea about how it work with test  `built-ins/Function/internals/Construct`

**Comment by hermanventer** [comment]
I don't understand any of this. Why does ProxyValue have its own $Realm property? I would think that just using the property in Value would be equivalent. Can anyone educate me on this?

**Comment by Kingwl** [comment]
i guess because of in `GetFunctionRealm`  

>  1. Assert: obj is a callable object.
> 2. If obj has a [[Realm]] internal slot, then
> a. Return obj.[[Realm]].
> 3. If obj is a Bound Function exotic object, then
> a. Let target be obj.[[BoundTargetFunction]].
> b. Return ? GetFunctionRealm(target). 4. If obj is a Proxy exotic object, then
> a. If obj.[[ProxyHandler]] is null, throw a TypeError exception. b. Let proxyTarget be obj.[[ProxyTarget]].
> c. Return ? GetFunctionRealm(proxyTarget). 5. Return the current Realm Record.

File: src/values/ProxyValue.js [hunk 1]
```diff
@@ -44,10 +44,14 @@ export default class ProxyValue extends ObjectValue {
 
   static trackedPropertyNames = ObjectValue.trackedPropertyNames.concat(["$ProxyTarget", "$ProxyHandler"]);
 
-  getTrackedBindings(): Array<string> {
+  getTrackedPropertyNames(): Array<string> {
```
Comment (hermanventer) [inline src/values/ProxyValue.js]: This part is a fix we should keep.

**Comment by hermanventer** [comment]
OK, I've poured over the specification and tried to understand what [[Realm]] is all about. I can't say that it is much clearer now, but it does seem to me that Prepack is significantly different from the spec in this regard, since all Prepack values have a $Realm property, whereas only Function values have a [[Realm]] property in the specification.

Ideally, the Realm class should be renamed into something else and should live in an untracked property. Then a new Realm class should be created as per the spec and it's instances should live in FunctionValue instances, as well as the non-value, specification constructs. 

This will be a non trivial task, so for now, the best thing would be to abandon the introduction of getRealm and to change the definition of GetFunctionRealm so that step 2 moves to after step 4.

**Comment by hermanventer** [comment]
Also delete the assignment to $Realm and replace all references to "this.realm" with "this.$Realm" and delete the "realm" property from ProxyValue.

**Pull request closed without merging**